zebrad/src/commands/start.rs

```diff
@@ -40,27 +41,37 @@ pub struct StartCmd {
 
 impl StartCmd {
     async fn start(&self) -> Result<(), Report> {
-        info!(?self, "starting to connect to the network");
+        let config = app_config().clone();
+        info!(?config);
 
-        let config = app_config();
+        info!("initializing node state");
         let state = zebra_state::init(config.state.clone(), config.network.network);
+
+        info!("initializing chain verifier");
         let verifier = zebra_consensus::chain::init(
             config.consensus.clone(),
             config.network.network,
             state.clone(),
         )
         .await;
 
-        // The service that our node uses to respond to requests by peers
-        let node = Buffer::new(
-            service_fn(|req| async move {
-                debug!(?req, "inbound peer request");
-                Ok::<zebra_network::Response, Report>(zebra_network::Response::Nil)
-            }),
-            1,
-        );
-        let (peer_set, _address_book) = zebra_network::init(config.network.clone(), node).await;
+        info!("initializing network");
+
+        // The service that our node uses to respond to requests by peers. The
+        // load_shed middleware ensures that we reduce the size of the peer set
+        // in response to excess load.
+        let (setup_tx, setup_rx) = oneshot::channel();
+        let inbound = ServiceBuilder::new()
+            .load_shed()
+            .buffer(20)
+            .service(Inbound::new(setup_rx, state.clone()));
+
+        let (peer_set, address_book) = zebra_network::init(config.network.clone(), inbound).await;
+        setup_tx
+            .send((peer_set.clone(), address_book))
+            .map_err(|_| eyre!("could not send setup data to inbound service"))?;
```

Comment:
do we want to discard this source error? If not we can use `wrap_err` to more easily construct a `Report` from a string while retaining the source error.

```suggestion
            .wrap_err("could not send setup data to inbound service")?;
```

Reply:
nice, i didn't think to use `wrap_err`.

Reply:
Ah, the reason that `wrap_err` doesn't work here is that the oneshot error gives the value back if the send failed.  But the value we send in the oneshot isn't an error.

Reply:
aah, okay, makes sense. Might be good to put `_discarded` instead of `_` or something to make it clear that the discard is intentional so my brain doesn't light up a month from now when I see it again randomly and have forgotten why we didn't use wrap_err.